### PR TITLE
4 to 5

### LINKED ISSUE #871: Pose updates for nested models are incorrect
## Environment
* OS Version: Ubuntu
* Source: Built from latest bazel branches

## Description
A common model composition paradigm used in our org is to model robot arms and end-effectors separately and connect them 
together with fixed joints, as such:
```
<model name = "parent_model">
    <link name="parent_link">...</link>
    <include><uri>model://child</uri> <name> child_model </name></include>
    <joint name="parent_joint">
        <parent> parent_model </parent>
        <child> parent_model </child>
    </joint>
</model>
```
However, pose updates seem to be wrong for the included model since updating to Ignition Edifice. I have a simple test sdf file with two double pendulums (video attached). Both pendulums are planar and have the base link fixed to the world, followed by revolute joint - link - revolute joint - link. The left pendulum is modeled with a nested model, i.e.  model{revolute joint - link} - revolute joint - link, whereas the right pendulum is modeled as a flat list.

The link poses are correct for the right pendulum, whereas they are wrong for some of the links for the left nested pendulum.

## Steps to reproduce
```
blaze run -c opt ign_tools:ign -- gazebo -r -v 4 test_pendulums.txt
```

## Output
[test_pendulums.txt](https://github.com/ignitionrobotics/ign-gazebo/files/6689063/test_pendulums.txt)
https://user-images.githubusercontent.com/2412842/122818055-92cc0d00-d28d-11eb-926c-ccdc9a169f4c.mov
Screen Shot 2021-06-21 at 12 39 20 PM

### PR DESCRIPTION
**Pull request by adlarkin**
# ➡️ Forward port

Port `ign-gazebo4` to `ign-gazebo5`

Branch comparison: https://github.com/ignitionrobotics/ign-gazebo/compare/ign-gazebo5...ign-gazebo4

Part of this forward-port involved fixing merge conflicts for #876. I also updated the physics system to handle the scenario where a parent model and nested model each have their own canonical links, but only the canonical link of the parent model moves. In this scenario, although the nested model did not move, the nested model and all of its links needs to be updated since the nested model pose in the physics system is saved w.r.t. the parent model. This scenario did not need to be addressed in #876 since `ign-gazebo4` does not have full nested model support. I believe that with this forward port and additional fix, we can close #871 - cc @azeey 

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: examples/plugin/hello_world/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.10.2 FATAL_ERROR)
+
+find_package(ignition-cmake2 REQUIRED)
+
+project(Hello_world)
+
+ign_find_package(ignition-plugin1 REQUIRED COMPONENTS register)
+set(IGN_PLUGIN_VER ${ignition-plugin1_VERSION_MAJOR})
+
+ign_find_package(ignition-gazebo4 REQUIRED)
```
Comment (chapulina) [inline examples/plugin/hello_world/CMakeLists.txt:10]:
```suggestion
ign_find_package(ignition-gazebo5 REQUIRED)
```

Also below

**Review by chapulina — COMMENTED** [review]
LGTM, but I didn't look closely at the changes from #876. It would be nice if @azeey could take a look.

**Review by azeey — APPROVED** [review]
LGTM with passing `INTEGRATION_examples_build`

**Comment by adlarkin** [comment]
`INTEGRATION_examples_build` is passing, so I will go ahead and merge this

**Pull request merged**